code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -60,10 +59,73 @@
 	return NONE
 
 
+/obj/machinery/chem_heater/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		. += span_notice("The status display reads: Heating reagents at <b>[heater_coefficient * 1000]%</b> speed.")
+		if(!QDELETED(beaker))
+			. += span_notice("It has a beaker of [beaker.reagents.total_volume] units capacity.")
+			if(beaker.reagents.is_reacting)
+				. += span_notice("Its contents are currently reacting.")
+		else
+			. += span_warning("There is no beaker inserted.")
+		. += span_notice("Its heating is turned [on ? "On" : "Off"].")
+		. += span_notice("The status display reads: Heating reagents at <b>[heater_coefficient * 1000]%</b> speed.")
+		if(panel_open)
+			. += span_notice("Its panel is open and can now be [EXAMINE_HINT("pried")] apart.")
+		else
+			. += span_notice("Its panel can be [EXAMINE_HINT("pried")] open")
+
 /obj/machinery/chem_heater/update_icon_state()
 	icon_state = "[base_icon_state][beaker ? 1 : 0]b"
 	return ..()
 
+/obj/machinery/chem_heater/wrench_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(default_unfasten_wrench(user, tool) == SUCCESSFUL_UNFASTEN)
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/chem_heater/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(default_deconstruction_screwdriver(user, "mixer0b", "[base_icon_state][beaker ? 1 : 0]b", tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/chem_heater/crowbar_act(mob/living/user, obj/item/tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/chem_heater/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == beaker)
+		UnregisterSignal(beaker.reagents, COMSIG_REAGENTS_REACTION_STEP)
+		beaker = null
+		update_appearance()
+
+/obj/machinery/chem_heater/RefreshParts()
+	. = ..()
+	heater_coefficient = 0.1
+	for(var/datum/stock_part/micro_laser/micro_laser in component_parts)
+		heater_coefficient *= micro_laser.tier
+
+/obj/machinery/chem_heater/attackby(obj/item/held_item, mob/user, params)
+	if((held_item.item_flags & ABSTRACT) || (held_item.flags_1 & HOLOGRAM_1))
+		return ..()
+
+	if(beaker)
+		if(istype(held_item, /obj/item/reagent_containers/dropper) || istype(held_item, /obj/item/reagent_containers/syringe))
+			var/obj/item/reagent_containers/injector = held_item
+			injector.afterattack(beaker, user, proximity_flag = TRUE)
+			return TRUE
+
+	if(is_reagent_container(held_item)  && held_item.is_open_container())
+		if(replace_beaker(user, held_item))
+			ui_interact(user)
+		balloon_alert(user, "beaker added!")
+		return TRUE
```

Comment:
Can you move these non-combat interactions to `item_interaction` rather than `attackby`? It'd be a huge help

Reply:
Interesting, yup I'll do that. But Can you give an example of an custom combat interaction inside attackby().

Because in that case we should make this change throughout the code base and make attackby() non overridable?

Because attackby() will allow you to hit any object with anything by default and if we make it just for combat purposes then our job is done with this proc and we can lock it in? Or is there an case where we can make an custom combat interaction?.

Again nit related to this PR I'm just trying to learn what is possible

Reply:
Hmm wait. 

Let's say A attacks B

does item_interact() get called on both A and B?

or does item_interact() get called only on B and attackby() get's called on both A and B?

Reply:
![285695709-bf5cc4bb-5a1f-42e3-921d-9a57bc6096cc](https://github.com/tgstation/tgstation/assets/51863163/fce8007d-9d2e-46be-a45a-5ab724b36919)

The intent is to put all non-combat related interactions between items and atoms in `interact_with_atom` (if on the item itself) or `item_interaction` (if on the atom) 

Item interactions being things like: Using a tool on something, putting a beaker in a machine, scanning someone with a health scanner, attaching a medal to someone's chest

Combat interactions being like: The supermatter sword dusting things it hits, training dummies which recoil when smacked

Reply:
I see. Ideally we should divide all interactions into 2 procs

1) Check if an action can be performed.(From your diagram a good example would be `pre_attack()`, & `pre_attack_secondary()` which should be pure in my opinion

2) Do the actual action `attackby()` & other procs

So by that logic we should also have procs like `can_item_interact()` or something like but some of these procs boundaries are so blurred & do both that it's too hard at this point to seperate them now.

Anyway by your principle i could make a future PR to move all non combat interactions to `item_interact_with()` so you have given me a good idea. Thanks for the explanation